src/serialport_win.cpp

```diff
@@ -311,6 +283,72 @@ void EIO_Write(uv_work_t* req) {
   } while (data->bufferLength > data->offset);
 }
 
+void EIO_Read(uv_work_t* req) {
+  ReadBaton* data = static_cast<ReadBaton*>(req->data);
+  data->bytesRead = 0;
+  int errorCode = ERROR_SUCCESS;
+
+  char* offsetPtr = data->bufferData;
+  offsetPtr += data->offset;
+
+  // Event used by GetOverlappedResult(..., TRUE) to wait for incoming data or timeout
+  // Event MUST be used if program has several simultaneous asynchronous operations
+  // on the same handle (i.e. ReadFile and WriteFile)
+  HANDLE hEvent = CreateEvent(NULL, TRUE, FALSE, NULL);
+
+  while (true) {
+    OVERLAPPED ov = {0};
+    ov.hEvent = hEvent;
+
+    // Start read operation - synchrounous or asynchronous
+    DWORD bytesReadSync = 0;
+    if (!ReadFile((HANDLE)data->fd, offsetPtr, data->bytesToRead, &bytesReadSync, &ov)) {
+      errorCode = GetLastError();
+      if (errorCode != ERROR_IO_PENDING) {
```

Comment:
Thanks. I'll keep my eye on it. Looks like someone's still able to repro with the proposed fix.